### ISSUE TITLE
Consider a PR for `.current` similar to Redis

### ISSUE THREAD

**Issue by harlow** [DataDog/dogstatsd-ruby#25]
We have a number of tags that we instantiate Statsd with. With `Redis.current` we do something similar:

```ruby
# config/initializers/redis.rb
config = {
  # ...
}
Redis.current = Redis.new(config)
```

https://github.com/redis/redis-rb/blob/b9ac235fa03de173d1d382878757850afae94bc3/lib/redis.rb#L19-L25

Would DataDog team consider a PR that does same for Statsd?

```ruby
# config/initializers/redis.rb
tags: {
  # env, service name, revision, etc
}
Statsd.current = Statsd.new(url, port, tags)
```

**Comment by yannmh** [comment]
Thank @harlow 

Could you give us more details about the use cases please ? I am not sure to understand how it would help you.

-- Yann

**Comment by harlow** [comment]
hi @yannmh we use `Statsd` throughout our codebase so it would be nice to be able to reference an initialized version of it.

```ruby
# config/initializers/statsd.rb
tags: {
  # env, service name, revision, etc
}
Statsd.current = Statsd.new(url, port, tags)
```

```ruby
# app/models/foo.rb
class Foo
  def bar
    if some_success_case?
      statsd.increment('xyz')
    end
  end

  private

  def statsd
    Statsd.current
  end
end
```

This was we have access to the configured `Statsd.new` that was initialized in config.

**Comment by yannmh** [comment]
Seems like a nice feature, thanks for the further explanations @harlow ! 
We'll certainly consider it for an upcoming release.

**Comment by sj26** [comment]
@harlow the easiest way we've found to do this is set a global variable for use across your application:

```ruby
$statsd = Datadog::Statsd.new
```

**Comment by aharpervc** [comment]
Is using a global variable compatible with the v5 requirement of needing to call `close`? Or, put another way, how could you do the global variable approach in v5?

I'm trying to upgrade an app that used a similar pattern, but I want to use buffer_flush_interval and not send stats live for each invocation of `increment` (etc).

Thoughts?

**Comment by ivoanjo** [comment]
You could perhaps use [`Kernel#at_exit`](https://rubyapi.org/3.1/o/kernel#method-i-at_exit) to ask Ruby to call it automatically on shutdown, although there may be a better way depending on your application.

**Comment by aharpervc** [comment]
> You could perhaps use [`Kernel#at_exit`](https://rubyapi.org/3.1/o/kernel#method-i-at_exit) to ask Ruby to call it automatically on shutdown, although there may be a better way depending on your application.

Yep, that's more or less what I figured out over here (then ran into a different issue): https://github.com/DataDog/dogstatsd-ruby/issues/265#issue-1384261981
